File: guides/micronaut-graalpy/java/src/test/java/example/micronaut/HelloControllerTest.java
package example.micronaut;

import io.micronaut.http.HttpRequest;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.runtime.server.EmbeddedServer;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest // <1>
public class HelloControllerTest {

    @Inject
    EmbeddedServer server;

    @Inject
    @Client("/")  // <2>
    HttpClient client;

    @Test
    void testHelloResponse() {
        String response = client.toBlocking().retrieve(HttpRequest.GET("/hello")); // <3>
        assertEquals("Hello World", response);
    }

}

File: guides/hello-base/kotlin/src/test/kotlin/example/micronaut/HelloControllerTest.kt
package example.micronaut

import io.micronaut.http.HttpRequest
import io.micronaut.http.client.HttpClient
import io.micronaut.http.client.annotation.Client
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import jakarta.inject.Inject

@MicronautTest // <1>
class HelloControllerTest(@Client("/") val client: HttpClient) { // <2>

    @Test
    fun testHello() {
        val request: HttpRequest<Any> = HttpRequest.GET("/hello")  // <3>
        val body = client.toBlocking().retrieve(request)
        assertNotNull(body)
        assertEquals("Hello World", body)
    }
}
